src/cryptonote_basic/cryptonote_basic_impl.cpp

```diff
@@ -86,11 +86,12 @@ namespace cryptonote {
     return CRYPTONOTE_MAX_TX_SIZE;
   }
   //-----------------------------------------------------------------------------------------------
-  bool get_block_reward(size_t median_size, size_t current_block_size, uint64_t already_generated_coins, uint64_t &reward, uint8_t version) {
+  bool get_block_reward(size_t median_size, size_t current_block_size, uint64_t already_generated_coins, uint64_t &reward, uint8_t version, size_t height) {
     static_assert(DIFFICULTY_TARGET_V2%60==0&&DIFFICULTY_TARGET_V1%60==0,"difficulty targets must be a multiple of 60");
-    const int target = version < 2 ? DIFFICULTY_TARGET_V1 : DIFFICULTY_TARGET_V2;
+    const int target = version < 2 && height < HARDFORK_1_HEIGHT ? DIFFICULTY_TARGET_V1 : DIFFICULTY_TARGET_V2;
     const int target_minutes = target / 60;
-    const int emission_speed_factor = EMISSION_SPEED_FACTOR_PER_MINUTE - (target_minutes-1);
+    assert(target_minutes & (target_minutes - 1) == 0);     // assume target_minutes to be power of 2
+    const int emission_speed_factor = EMISSION_SPEED_FACTOR_PER_MINUTE - (int)std::log2(target_minutes);
```

Comment:
This might be OK in practice, but I'd make my own that works on powers of 2 instead of using a floating point lib function.

Reply:
ok